py/sight/sight.py

```diff
@@ -986,47 +987,47 @@ def run_generic_worker(
-        client_id=client_id,
-        worker_id=f'client_{client_id}_worker_{worker_location}',
-        question_label=sight_params['label'])
-    response = service.call(
-        lambda s, meta: s.WorkerAlive(req, 300, metadata=meta))
-    logging.info('Response from WorkerAlive RPC: %s', response)
-    if (response.status_type ==
-        service_pb2.WorkerAliveResponse.StatusType.ST_DONE):
-      break
-    elif (response.status_type ==
-          service_pb2.WorkerAliveResponse.StatusType.ST_RETRY):
-      # logging.info('Retrying in 5 seconds......')
-      # time.sleep(5)
-      backoff_interval *= 2
-      time.sleep(random.uniform(backoff_interval / 2, backoff_interval))
-      logging.info('backed off for %s seconds... and trying for %s',
-                   backoff_interval, num_retries)
-      num_retries += 1
-      if (num_retries >= 50):
+  with Sight.create(sight_params) as sight:
+    sight.widget_decision_state['num_decision_points'] = 0
+
+    optimizer = decision.Optimizer()
+    optimizer.obj = decision.setup_optimizer(sight, FLAGS.optimizer_type)
+    client_id, worker_location = decision._configure_client_and_worker(
+        sight=sight)
+    num_retries = 0
+    backoff_interval = 0.5
+    while True:
+      # #? new rpc just to check move forward or not?
+
+      req = service_pb2.WorkerAliveRequest(
+          client_id=client_id,
+          worker_id=f'client_{client_id}_worker_{worker_location}',
+          question_label=sight_params['label'])
+      response = service.call(
+          lambda s, meta: s.WorkerAlive(req, 300, metadata=meta))
+      logging.info('Response from WorkerAlive RPC: %s', response)
+      if (response.status_type ==
+          service_pb2.WorkerAliveResponse.StatusType.ST_DONE):
         break
-    elif (response.status_type ==
-          service_pb2.WorkerAliveResponse.StatusType.ST_ACT):
-      process_worker_action(response, sight, driver_fn, sight_params['label'],
-                            optimizer.obj)
-    else:
-      raise ValueError('Invalid response from server')
+      elif (response.status_type ==
+            service_pb2.WorkerAliveResponse.StatusType.ST_RETRY):
+        # logging.info('Retrying in 5 seconds......')
+        # time.sleep(5)
+        backoff_interval *= 2
+        time.sleep(random.uniform(backoff_interval / 2, backoff_interval))
+        logging.info('backed off for %s seconds... and trying for %s',
+                    backoff_interval, num_retries)
+        num_retries += 1
+        if (num_retries >= 50):
+          break
+      elif (response.status_type ==
+            service_pb2.WorkerAliveResponse.StatusType.ST_ACT):
+        process_worker_action(response, sight, driver_fn, sight_params['label'],
+                              optimizer.obj)
+      else:
+        raise ValueError('Invalid response from server')
 
-    logging.info('Exiting the training loop.')
+      logging.info('Exiting the training loop.')
```

Comment:
It looks like this message will now be printed in every iteration. Please move to after the loop